src/mrpro/data/traj_calculators/_KTrajectoryCalculator.py

```diff
@@ -57,18 +58,17 @@ def _kfreq(self, kheader: KHeader) -> torch.Tensor:
         ValueError
             Center sample has to be the same for each readout
         """
-        n_samples = kheader.acq_info.number_of_samples
+        n_samples = torch.unique(kheader.acq_info.number_of_samples)
         center_sample = kheader.acq_info.center_sample
-
-        # Verify that each readout has the same number of samples and same center sample
-        if len(torch.unique(n_samples)) > 1:
+        if len(n_samples) > 1:
             raise ValueError('Trajectory can only be calculated if each acquisition has the same number of samples')
-        if len(torch.unique(center_sample)) > 1:
-            raise ValueError('Trajectory can only be calculated if each acquisition has the same center sample')
+        n_k0 = int(n_samples.item())
 
-        # Calculate points along readout
-        n_k0 = int(n_samples[0, 0, 0])
-        k0 = torch.linspace(0, n_k0 - 1, n_k0, dtype=torch.float32) - center_sample[0, 0, 0]
+        # Data can be obtained with standard or reversed readout (e.g. bipolar readout).
+        k0 = torch.linspace(0, n_k0 - 1, n_k0, dtype=torch.float32) - center_sample
+        # Data can be obtained with standard or reversed readout (e.g. bipolar readout).
+        reversed_readout_mask = (kheader.acq_info.flags[..., 0] & AcqFlags.ACQ_IS_REVERSE.value).bool()
```

Comment:
Just a question unrelated to the code: can `ACQ_IS_REVERSE ` be set using pulseq labels? Or would this only work with an additional meta ismrmrd file with pulseq sequences?

Reply:
Not sure what pulseq can set but this is based on a standard Siemens raw data flag, similiar to calibration data and so on.

Reply:
Do we need this if we use a pulseq trajectory calculator?

Otherwise, I think it will not be set...

Reply:
@schuenke
In the pulseq reference manual, label extension section, the binary flag REV is mentioned

Can you try if setting that flag in pulseq to 1 sets the reversed acquisition flag in the ismrmrd?

Reply:
Can try it later today, yeah